lib/remote/objectqueryhandler.cpp

```diff
@@ -257,6 +260,46 @@ bool ObjectQueryHandler::HandleRequest(
 			if (!joinedObj)
 				continue;
 
+			Type::Ptr reflectionType = joinedObj->GetReflectionType();
+			Expression *permissionFilter;
+
+			if (grantedTypes.find(reflectionType->GetName()) == grantedTypes.end()) {
+				String permission = "objects/query/" + reflectionType->GetName();
+
+				if (!FilterUtility::HasPermission(user, permission, &permissionFilter)) {
+					// The API user isn't authorized to access this relation, so just ignore it because there
+					// is no reason to raise an exception here since the actual object requested by the client
+					// has been found!!
+					continue;
+				}
+
+				grantedTypes.insert(reflectionType->GetName());
+			}
+
+			if (!resolvedObjs || !resolvedObjs->Contains(joinedObj)) {
+				Namespace::Ptr frameNS = new Namespace();
+				ScriptFrame permissionFrame(false, frameNS);
+
+				/*
+				 * We're going to evaluate the permission filters of the requested base object on all the other joined
+				 * relations, so when we have a filter like {{ match(..., "host.vars......" }}, then we won't be able to
+				 * access the host object from the namespace. Therefore, we have to register the base obj type to the
+				 * namespace before evaluating the filter for joined relations.
+				 */
+				frameNS->Set(type->GetName().ToLower(), obj);
```

Comment:
Please give (here in the comments) a realistic specific example of a permission with a filter which you make working via this line.

Reply:
This doesn't matter now! Just forgot to delete it.